### ISSUE TITLE
CSS @import url for loading fonts doesnt work

### ISSUE THREAD

**Issue by jaemil** [vercel/next.js#30567]
### What version of Next.js are you using?

12.0.1

### What version of Node.js are you using?

17.0.1

### What browser are you using?

Firefox, Chrome

### What operating system are you using?

Windows 10

### How are you deploying your application?

next start

### Describe the Bug

App.scss:
`@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;600;700&display=swap');`

Loading the font with next dev works but not with next build (next start).

Tested with `optimizeFonts: false` but with same result.

### Expected Behavior

In next 11.0.1 the font was imported correctly

### To Reproduce

Inside App.scss
`@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;600;700&display=swap');`

1. next build
2. next start

**Comment by alaadahmed** [comment]
Same here, I use TailwindCSS and in `globals.css` I do this: 
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  @import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700&display=swap');

  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }

  html,
  body {
    font-family: 'Poppins', sans-serif;
    line-height: 1.8;
  }
  // ...
```
 I see in chrome inspector that computed font is the one I imported, but it doesn't render as should be

On `next 11.1.2` this works fine!

**Comment by gthb** [comment]
@timneutkens I found the proximate cause: the `@import"..."` statement that should load the font appears way down in the middle of a CSS chunk. Per [the specification](https://drafts.csswg.org/css-cascade/#at-import):

> Any @import rules must precede all other valid at-rules and style rules in a style sheet (ignoring @charset), or else the @import rule is invalid.

Sure enough, if I manually edit the chunk to put this `@import` statement at the top, the font loads correctly.

**Comment by gthb** [comment]
Possibly because the `@import"..."` statement has been so eagerly minimized as to have no space between the `@import` and the `"'..."` but [`css-loader` expects a space](https://github.com/webpack-contrib/css-loader/blob/9266240be2a09cc892c5ae26237e936f2e6f0c0b/lib/parseSource.js#L205-L209)?
```
https://github.com/webpack-contrib/css-loader/blob/9266240be2a09cc892c5ae26237e936f2e6f0c0b/lib/parseSource.js#L205-L209
```
'@\\s*import\\s+"([^"]*)"\\s*([^;\\n]*);': importMatch,
"@\\s*import\\s+'([^'']*)'\\s*([^;\\n]*);": importMatch,
'@\\s*import\\s+url\\s*\\(\\s*"([^"]*)"\\s*\\)\\s*([^;\\n]*);': importMatch,
"@\\s*import\\s+url\\s*\\(\\s*'([^']*)'\\s*\\)\\s*([^;\\n]*);": importMatch,
"@\\s*import\\s+url\\s*\\(\\s*([^)]*)\\s*\\)\\s*([^;\\n]*);": importMatch,
```

**Comment by mrodrigues95** [comment]
Adding a custom `_document.js` and injecting it with the desired font as @alaadahmed suggested fixes it for me as well.

**Comment by gthb** [comment]
Yeah, the workaround of an explicit `<link>` tag in `<Head>` in `_document.js` works for my case as well — but only once I also disable [font optimization](https://nextjs.org/docs/basic-features/font-optimization) with `optimizeFonts: false` in `next.config.js` (else the `<link>` tag gets its `href` rewritten to `data-href` so the browser ignores it).

Thanks!

**Comment by petehl** [comment]
> * include `rel="stylesheet"` in the `<link>` tag, else it only worked in dev mode, not in the production build (don't know if that's for some reason particular to my project)

Including the `rel="stylesheet"`was what I was missing as well. 

So adding `<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Poppins...." />` as child for `<Head>` in `_document.tsx` had our fonts rendering again.

**Comment by sveggiani** [comment]
Are there any news on this?

**Comment by sveggiani** [comment]
Are there any news or progress on this?

**Comment by alizahid** [comment]
@sveggiani It's to be released in the 12.0.4 milestone, which is scheduled for November 30. Shouldn't be too long now.

**Comment by zackdotcomputer** [comment]
Unfortunately, not fixed in Next 12.0.4 or 12.0.5 canary 3

**Comment by nratter** [comment]
I'm on 12.0.6 and this is still not working for me.

**Comment by joshdocking** [comment]
Importing Typekit fonts is still broken for me in 12.0.7. Should this issue still be open?

**Comment by balazsorban44** [comment]
No. Please open a separate issue with a reproduction.

**Comment by balazsorban44** [comment]
There is a possibility that this is the same as #32645
